src/compiler/checker.ts

```diff
@@ -1526,17 +1534,53 @@ module ts {
                     }
                 }
 
+                function writeSymbolTypeReference(symbol: Symbol, typeArguments: Type[], pos: number, end: number) {
+                    if (!isReservedMemberName(symbol.name)) {
+                        buildSymbolDisplay(symbol, writer, enclosingDeclaration, SymbolFlags.Type);

```

Comment:
I think instead of `SymbolFlags.Type` you should pass `symbol.flags`. Though I admit, I don't know what it will do, and I'm really not sure. I'm just thinking in the case where it is a function with type parameters, it is not really a Type.